app/src/test/java/io/apicurio/registry/content/ContentCanonicalizerFactoryTest.java

```diff
@@ -171,5 +171,44 @@ void testKafkaConnect() {
         Assertions.assertEquals(expected, actual);
     }
 
+    /**
+    * Test method for {@link io.apicurio.registry.content.ContentCanonicalizerFactory#create(io.apicurio.registry.types.ArtifactType)}.
+    */
+    @Test
+    void testXsd() {
+       ContentCanonicalizerFactory factory = new ContentCanonicalizerFactory();
+       ContentCanonicalizer canonicalizer = factory.create(ArtifactType.XSD);
+       String before = "<xs:schema xmlns:xs=\"http://www.w3.org/2001/XMLSchema\">\n"
+               + "<xs:element             name=\"shiporder\">\n" + "  <xs:complexType>\n"
+               + "    <xs:sequence>\n"
+               + "      <xs:element            name=\"orderperson\" type=\"xs:string\"/>\n"
+               + "      <xs:element       name=\"shipto\">\n" + "        <xs:complexType>\n"
+               + "          <xs:sequence>\n"
+               + "            <xs:element          type=\"xs:string\"  name=\"name\" />\n"
+               + "            <xs:element               type=\"xs:string\" name=\"address\"/>\n"
+               + "            <xs:element      type=\"xs:string\"  name=\"city\"/>\n"
+               + "            <xs:element    type=\"xs:string\" name=\"country\"/>\n"
+               + "          </xs:sequence>\n" + "        </xs:complexType>\n" + "      </xs:element>\n"
+               + "    </xs:sequence>\n"
+               + "    <xs:attribute use=\"required\" type=\"xs:string\" name=\"orderid\"/>\n"
+               + "  </xs:complexType>\n" + "</xs:element>\n" + "</xs:schema>";
 
+       String expected = "<xs:schema xmlns:xs=\"http://www.w3.org/2001/XMLSchema\">\n"
+               + "<xs:element name=\"shiporder\">\n" + "  <xs:complexType>\n" + "    <xs:sequence>\n"
+               + "      <xs:element name=\"orderperson\" type=\"xs:string\"></xs:element>\n"
+               + "      <xs:element name=\"shipto\">\n" + "        <xs:complexType>\n"
+               + "          <xs:sequence>\n"
+               + "            <xs:element name=\"name\" type=\"xs:string\"></xs:element>\n"
+               + "            <xs:element name=\"address\" type=\"xs:string\"></xs:element>\n"
+               + "            <xs:element name=\"city\" type=\"xs:string\"></xs:element>\n"
+               + "            <xs:element name=\"country\" type=\"xs:string\"></xs:element>\n"
+               + "          </xs:sequence>\n" + "        </xs:complexType>\n" + "      </xs:element>\n"
+               + "    </xs:sequence>\n"
+               + "    <xs:attribute name=\"orderid\" type=\"xs:string\" use=\"required\"></xs:attribute>\n"
+               + "  </xs:complexType>\n" + "</xs:element>\n" + "</xs:schema>";
```

Comment:
Can you externalize these two files to test resources?

Reply:
👍

Reply:
Is it ok to extend `AbstractRegistryTestBase` to utilise the methods there?

Reply:
Yes I think so.